erpnext/manufacturing/doctype/production_plan/production_plan.py

```diff
@@ -483,6 +483,35 @@ def get_exploded_items(item_details, company, bom_no, include_non_stock_items, p
 			item_details.setdefault(d.get('item_code'), d)
 	return item_details
 
+def get_exploded_items_with_subassembly(item_details, company, bom_no, include_non_stock_items, planned_qty=1):
+	explosion_items = get_items_for_explod_and_subassembly("BOM Explosion Item", company, bom_no, include_non_stock_items, planned_qty)
+	bom_items = get_items_for_explod_and_subassembly("BOM Item", company, bom_no, include_non_stock_items, planned_qty)
+
+	for d in explosion_items + bom_items:
+		item_details.setdefault(d.get('item_code'), d)
+	return item_details
+
+def get_items_for_explod_and_subassembly(item, company, bom_no, include_non_stock_items, planned_qty=1):
```

Comment:
```suggestion
def get_items_to_explode_with_subassembly(item, company, bom_no, include_non_stock_items, planned_qty=1):
```
Also add a docstring.